src/math.ts

```diff
@@ -258,22 +284,38 @@ export const calculateSqrtPriceAfterSlippage = (
     return sqrtPrice
   }
 
-  const multiplier = toPercentage(1n) + (up ? slippage : -slippage)
+  const multiplier = getPercentageDenominator() + (up ? slippage : -slippage)
   const price = sqrtPriceToPrice(sqrtPrice)
-  const priceWithSlippage = price * multiplier * toPercentage(1n)
-  const sqrtPriceWithSlippage = priceToSqrtPrice(priceWithSlippage) / toPercentage(1n)
+  const priceWithSlippage = price * multiplier * getPercentageDenominator()
+  const sqrtPriceWithSlippage = priceToSqrtPrice(priceWithSlippage) / getPercentageDenominator()
 
   return sqrtPriceWithSlippage
 }
 
 export const toLiquidity = (value: bigint, offset = 0n) => {
+  assert(offset < LiquidityScale, 'offset must be less than LiquidityScale')
   return value * 10n ** (LiquidityScale - offset)
 }
 
-export const toSqrtPrice = (value: bigint, offset = 0n) => {
+export const toSqrtPrice = (value: bigint, offset = 0n): bigint => {
+  assert(offset < SqrtPriceScale, 'offset must be less than SqrtPriceScale')
```

Comment:
I am not sure if you have correct import for `assert`. It might be better to display error with number insted on sqrtPriceScale
```
assert(offset < SqrtPriceScale, `offset must be less than ${SqrtPriceScale}`)
```